src/component/Pagination.jsx

```diff
@@ -1,24 +1,95 @@
 // Pagination.js
-const Pagination = ({ pagebuttons, activePage, handlePageClick }) => {
+import styled from "styled-components";
+import { useState } from "react";
+
+const PaginationContainer = styled.div`
+  display: flex;
+  margin: 0 auto;
+  width: 304px;
+  height: 40px;
+  align-items: center;
+`;
+
+const PaginationButton = styled.button`
+  border-radius: 50px;
+  width: 40px;
+  height: 40px;
+  background-color: #ffffff;
+  border: 1px solid #e5e7e8;
+  margin-left: 4px;
+  margin-top: 40px;
+  font-size: 16px;
+  line-height: 26px;
+  font-weight: 600;
+
+  &:hover {
+    background-color: #f0f0f0;
+    border-color: #ccc;
+  }
+`;
+
+const Pagination = ({ activePage, setActivePage, totalpage }) => {
+  const [pagebuttons, setPagebuttons] = useState([1, 2, 3, 4, 5]);
+
+  const handleNextClick = () => {
+    console.log("totalpage:", totalpage);
+    if (activePage >= totalpage) {
+      return;
+    }
+    if (activePage % 5 === 0) {
+      pagebuttons.forEach((num, index) => {
```

Comment:
P2:
forEach에서 사용하지 않는 args의 경우 _로 선언하시면 사용하지 않는 값이라는 것을 좀더 명시적으로 알려줄 수 있습니다.

```suggestion
      pagebuttons.forEach((_, index) => {
```